include/boost/geometry/util/numeric_cast.hpp

```diff
@@ -0,0 +1,54 @@
+// Boost.Geometry
+
+// Copyright (c) 2024 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_UTIL_NUMERIC_CAST_HPP
+#define BOOST_GEOMETRY_UTIL_NUMERIC_CAST_HPP
+
+#include <boost/numeric/conversion/cast.hpp>
+#include <boost/rational.hpp>
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail
+{
+
+/// brief calls numeric cast
+template <typename Target, typename Source>
+struct numeric_caster
+{
+    static inline Target apply(Source const& source)
+    {
+        return boost::numeric_cast<Target>(source);
+    }
+};
+
+// Specialization for Boost.Rational
+template <typename Target, typename T>
+struct numeric_caster<Target, rational<T>>
+{
+   static inline Target apply(rational<T> const& source)
+    {
+        return boost::rational_cast<Target>(source);
+    }
+};
+
+template <typename Target, typename Source>
+inline Target numeric_cast(Source const& source)
```

Comment:
Why not put it in `bg::util` namespace with the rest of utilities?

Reply:
I noticed that `coordinate_cast` was also in `detail` so I put it there.
Also it is not really meant for end users (but `util` neither).
I didn't think of `util` - yes, maybe it is better there. I'll change it tomorrow.

Reply:
✅ changed it to `util::`